paddle/parameter/ParameterUpdaterHook.cpp

```diff
@@ -156,20 +226,30 @@ class StringIntPairHasher {
 
 static WeakKVCache<std::pair<std::string, int>,
                    IParameterUpdaterHook,
-                   StringIntPairHasher>
-    g_hookCache_;
+                   StringIntPairHasher> g_hookCache_;
 
 /**
  * ParameterUpdaterHook actually factory method.
  */
 static IParameterUpdaterHook* createImpl(
     const ParameterUpdaterHookConfig& config) {
   auto& type = config.type();
-  if (type == "pruning") {
-    if (config.has_purning_mask_filename()) {
+  if (type == "pruning_static") {
```

Comment:
the specific 'pruning' type  is dynamic one, and 'pruning_static' is static which read mask from the file.